server/src/main/java/org/opensearch/action/support/replication/ReplicationOperation.java

```diff
@@ -99,6 +100,7 @@
     private final TimeValue initialRetryBackoffBound;
     private final TimeValue retryTimeout;
     private final long primaryTerm;
+    private final boolean forceReplicationIfRemoteTranslogEnabled;
```

Comment:
@reta this is an action item already. I am trying to have some generic terms here and not tie with remote translog. This way we would have levers to allow a world where we could have a mix of local & remote replication. Post that change, this should be free from remote translog reference.

Reply:
@reta I share the same concern see my comments https://github.com/opensearch-project/OpenSearch/pull/5282#discussion_r1028390527 and https://github.com/opensearch-project/OpenSearch/pull/5282#discussion_r1038376499

Reply:
@reta @Bukhtawar  have remodelled the LLD and now there very fewer occurrence of remote translog across the code base. Let me know if there is any other feedback around this.

Reply:
Wondering why can't we simply have TransportActions override and provide their own ReplicationProxy rather than modifying the constructor argument all over the place.

Reply:
Firstly, modifying constructor of ReplicationOperation is required as we would need to pass either the override policy or the replication proxy. And reason for doing so - is that TransportReplicationAction internally creates an object of ReplicationOperation per request and ReplicationOperation is the class where we have the performOnReplica method. There is only this class where constructor has been modified (since there is no other way). For TransportReplicationAction, the constructor has not been modified. 

Couple of reasons on why ReplicationProxy has been created in the ReplicationOperation constructor through the ReplicationProxyFactory -
1. Technical challenge - currently performOnReplica is method provided as a ReplicationOperation feature which is common across different TransportReplicationAction. the performOnReplica method accesses a bunch of instance variable of ReplicationOperation class. Technically, it is very challenging to know what is the performOnReplica method going to do from outside ReplicationOperation class without it's object. If we were to take out performOnReplica method out, it would required many methods as well to be removed from the ReplicationOperation class.
2. Functional reason - The behavior of performOnReplica needs to be defined per replication mode and not per action per replication mode. Even today, the performOnReplica method is universal for all TransportReplicationActions. There is already a ReplicasProxy class present which we can use for overriding different behavior present. But pont to note is - that we only want to define behavior of performOnReplica per replication mode.

Reply:
Please update the PR based on our offline discussion

Reply:
@Bukhtawar done.